ACPs/194-streaming-asynchronous-execution/README.md

```diff
@@ -0,0 +1,394 @@
+
+## Security Considerations
+
+### Worst-case transaction validity
+
+To avoid a DoS vulnerability on execution, we require an upper bound on transaction gas cost (i.e. amount $\times$ price).
+Beyond regular requirements for transaction validity (e.g. nonce, signature, etc.) we therefore introduce "worst-case cost" validity.
+
+Consider a hypothetical situation in which all enqueued transactions use their entire gas limit during execution.
+We can prove that, for every transaction, this would result in the greatest possible:
+
+1. Consumption of gas units (by definition of the gas limit); and
+2. Gas excess $x$ (and therefore gas price) at the time of execution.
+
+For a queue of transactions $Q = \\{i\\}_ {i \ge 0}$ the gas excess $x_j$ immediately prior to execution of transaction $j \in Q$ is a monotonic, non-decreasing function of the gas usage of all preceding transactions in the queue; i.e. $x_j~:=~f(\\{g_i\\}_{i<j})$.
+
+To see this, consider transaction $0 \le k<j$ consuming gas $g_k$.
+A decrease in $g_k$ reduces the immediate increase of $x$.
+Furthermore, this lowered consumption can never reduce a chunk's gas surplus and hence any decrease of $x$ is $\ge$ predicted.
+The excess, and hence gas price, for every later transaction $x_{i>k}$ is therefore reduced:
+
+$$
+\downarrow g_k \implies
+\begin{cases}
+    \downarrow \frac{g \cdot (p-1)}{p} \\
+    \uparrow \frac{s}{p}
+\end{cases}
+\implies \downarrow \Delta x_k
+\implies \downarrow M \cdot \exp\left(\frac{x_{i>k}}{K}\right)
+$$
+
+Given maximal gas consumption under (1), the monotonicity of $f$ implies (2).
+
+Since we are working with non-negative integers, it follows that multiplying a transaction's gas limit by the hypothetical gas price of (2) results in its worst-case gas cost.
+Any sender able to pay for this upper bound (in addition to value transfers) is guaranteed to be able to pay for the actual execution cost.
+Transaction _acceptance_ under worst-case cost validity is therefore a guarantee of _settlement_.
+
+### Queue DoS protection
+
+Worst-case cost validity only protects against DoS at the point of execution but leaves the queue vulnerable to high-limit, low-usage transactions.
+For example, a malicious user could send a transfer-only transaction (21k gas) with a limit set to multiple chunks.
+Although they would have to have sufficient funds to theoretically pay for all the chunks, they would never actually be charged this amount. Pushing a sufficient number of such transactions to the queue would artificially inflate the worst-case cost of other users.
+
+Gas limits are typically set higher than expected gas used to allow for a buffer should gas estimates be imprecise.
+A lower bound of 50% of the specified limit, for example, would allow for a 100% buffer on estimates without penalising the sender, while still disincentivising falsely high limits.
+
+#### Upper bound on queue DoS
```

Comment:
This section should remain in the security consideration (and at the end we should plug in our planned parameters so that we can see the actual value on the worst case DoS multiplier)